backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/carousel/internal/BpkCarouselImpl.kt

```diff
@@ -0,0 +1,90 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.carousel.internal
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.rememberPagerState
+import kotlinx.coroutines.flow.distinctUntilChanged
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicator
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicatorStyle
+
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+internal fun BpkCarouselImpl(
+  modifier: Modifier = Modifier,
+  imageContent: @Composable (BoxScope.(Int) -> Unit),
+  currentImage: Int,
+  count: Int,
+  onImageChanged: ((Int) -> Unit)? = null,
+) {
+  Box(modifier = modifier) {
+    val initialPageIndex = (Int.MAX_VALUE / 2) + currentImage
+    val (selectedPage, setSelectedPage) = rememberSaveable { mutableStateOf(currentImage) }
+
+    val pagerState = rememberPagerState(initialPageIndex).also {
+      LaunchedEffect(it) {
+        snapshotFlow { it.currentPage }.distinctUntilChanged().collect { index ->
+          val page = getModNumber(index, count)
+          setSelectedPage(page)
+          onImageChanged?.invoke(page)
+        }
+      }
+    }
+
+    HorizontalPager(
```

Comment:
does this handle accessibility internally? ie allow switching between pages, announce the page change, count etc? how does it handle the `max_value` case?

Reply:
Unfortunately no, as currently the talkback would ignore the Carousel component

Reply:
Ok let me check what other platforms are doing re accessibility for this

Reply:
Just checked and the reason it is ignored is because we're not providing a content description for the items in the sample (the `image` content description is null) - can you set a content description please? this brings up a different issue where after each page change it switches out of the gallery. i wonder if that has to do with the infinite scroll logic

Reply:
I've tested removing different logic and indeed the count with MAX_VALUE causes this weird behaviour. 🙁
However, this is the suggested way to add infinite scroll the the Pager, I'm not sure if we could actually fix that.
https://github.com/google/accompanist/blob/main/sample/src/main/java/com/google/accompanist/sample/pager/HorizontalPagerLoopingSample.kt

Reply:
I'm gonna page @bvitaliyg here who knows a bit more about this than I.
I think in general we might need to change this a little more:
We have the `currentPage` in the API as per spec, but really it is treated like a `initialPage` property as the state is handled within the component. However I'm not sure how in this component this could be moved to the usage with the infinite scroll behaviour

Reply:
btw any idea why the label-check is stuck in the pipeline?

Reply:
good question! i noticed the same happened with another PR the other week. it looks like it didn't trigger. i just added/removed another label and that did it